src/sardana/macroserver/macros/standard.py

```diff
@@ -263,9 +263,15 @@ class set_lim(Macro):
     def run(self, motor, low, high):
         name = motor.getName()
         self.debug("Setting user limits for %s" % name)
-        motor.getPositionObj().setLimits(low, high)
-        self.output("%s limits set to %.4f %.4f (user units)" %
-                    (name, low, high))
+        if low > high:
+            raise Exception('set_lim exception: lower limit can not be a higher value than the high limit')
+        else:
+            try:
+                motor.getPositionObj().setLimits(low, high)
+                self.output("%s limits set to %.4f %.4f (user units)" %
+                (name, low, high))
+            except Exception as e:
+                self.debug("exception occured during setting user limits %s" % e)
```

Comment:
```suggestion
        if low > high:
            raise ValueError('lower limit must be lower than the higher limit')
        motor.getPositionObj().setLimits(low, high)
        self.output(
            "%s limits set to %.4f %.4f (user units)" % (name, low, high)
        )
```
- `else` is not necessary, an exception already controls the flow and one indentation level less is easier to read:)
- macros which due to some exceptional condition can not do what they are supposed to do should not catch the exception and return successfully. Let's imagine that we execute the macro in a sequence and the following macros rely on the previous macro result.
If you agree, could you please apply similar changes to the `set_lm` macro. Many thanks!